web/backend-fastapi/app/modules/agent/services/orchestrator_service.py

```diff
@@ -0,0 +1,170 @@
+import os
+import json
+from typing import Dict, List, Any, Optional
+from fastapi import HTTPException
+from fastapi.logger import logger
+from fastmcp import Client
+from fastmcp.exceptions import ToolError
+
+from ..agents import agent_registry
+from ..agents.orchestrator.orchestrator_agent import orchestrator_mcp
+from app.shared.models.azure import AzureAI
+from ..models.agent_model import AgentRequest, AgentResponse
+
+
+class OrchestratorService:
```

Comment:
It looks like this shares a lot of logic with the agent_service.py
I wonder if we could make a parent class that covers both of these.